src/psyclone/tests/psyclone_test_utils.py

```diff
@@ -338,3 +301,58 @@ def string_compiles(code, tmpdir, f90, f90flags):
             ofile.remove()
 
     return success
+
+
+# =============================================================================
+def get_invoke(algfile, api, idx=None, name=None):
+    '''
+    Utility method to get the idx'th or named invoke from the algorithm
+    in the specified file.
+    :param str algfile: name of the Algorithm source file (Fortran)
+    :param str api: which PSyclone API this Algorithm uses
+    :param int idx: the index of the invoke from the Algorithm to return
+                    or None if name is specified
+    :param str name: the name of the required invoke or None if an index
+                     is supplied
+    :returns: (psy object, invoke object)
+    :rtype: 2-tuple containing :py:class:`psyclone.psyGen.PSy` and
+            :py:class:`psyclone.psyGen.Invoke` objects.
+    :raises RuntimeError: if neither idx or name are supplied or if
+                          both are supplied
+    :raises RuntimeError: if the supplied name does not match an invoke in
+                          the Algorithm
+    '''
+    from psyclone.parse import parse
+    from psyclone.psyGen import PSyFactory
+
+    if (idx is None and not name) or (idx is not None and name):
+        raise RuntimeError("Either the index or the name of the "
+                           "requested invoke must be specified")
+
+    if api == "gocean1.0":
+        dir_name = "gocean1p0"
+    elif api == "dynamo0.3":
+        dir_name = "dynamo0p3"
+    elif api == "gocean0.1":
+        dir_name = "gocean0p1"
+    elif api == "dynamo0.1":
+        dir_name = "dynamo0p1"
+    elif api == "gunghoproto":
```

Comment:
Please remove support for "gunghoproto" here since the attempts to test it revealed more problems...

Reply:
Oh, you'll have to update the message for the `RuntimeError` below too (sorry).